src/heuristics.jl

```diff
@@ -0,0 +1,6 @@
+function thread_blocks_heuristic(len::Integer)
+    # TODO better threads default
+    threads = min(len, 256)
```

Comment:
Query supported threads from device?

Reply:
Well, we need to query for *optimal* thread count, which is not a straight forward query as far as I'm concerned.... I guess OpenCL has e.g. `CL_KERNEL_PREFERRED_WORK_GROUP_SIZE_MULTIPLE`, I'll see what CUDA has. (@maleadt, maybe?)

Reply:
Well occupancy isn't easy to predict, but there are af ew things here https://devblogs.nvidia.com/parallelforall/cuda-pro-tip-occupancy-api-simplifies-launch-configuration/

Reply:
For this to work, we probably need to make kernels full fledged objects with attached metadata and functions like the ones mentioned in that blog post. Currently not possible though.

Reply:
why? I mean, we could even have dummy object with a generated call function right now, or am I missing something?

Reply:
This change shouldn't change much, no?
Turn:
```Julia
kernel(x) = ...

@cuda (block, threads) kernel(x)
```
into 
```Julia
kernel(x) = ...
cu_kernel = CuKernel(kernel, x)
cu_kernel(x)
```
Or is the actual problem creating a CUDA kernel object equivalent to what they use in the blog (I've seen Kernel objects in CUDAdrv, so I'm assuming that that part is already done)?

Reply:
x-ref: https://github.com/JuliaGPU/CUDAnative.jl/issues/120